e2e: Undeploy fail is resources is not found

To clean up clusters after failed tests, we can use:

```console
% ./run.sh -test.run ///Undeploy
```

However it fails when some resource is not found:

```
2025-03-03T12:50:59.161+0200	ERROR	subscr-deploy-rbd-busybox	Failed to undeploy workload: placements.cluster.open-cluster-management.io "subscr-deploy-rbd-busybox" not found
2025-03-03T12:50:59.161+0200	ERROR	appset-deploy-rbd-busybox	Failed to undeploy workload: placements.cluster.open-cluster-management.io "appset-deploy-rbd-busybox" not found
2025-03-03T12:50:59.162+0200	ERROR	appset-deploy-cephfs-busybox	Failed to undeploy workload: placements.cluster.open-cluster-management.io "appset-deploy-cephfs-busybox" not found
2025-03-03T12:50:59.163+0200	ERROR	subscr-deploy-cephfs-busybox	Failed to undeploy workload: placements.cluster.open-cluster-management.io "subscr-deploy-cephfs-busybox" not found
...
--- FAIL: TestSuites (0.03s)
    --- PASS: TestSuites/Validate (0.03s)
        --- PASS: TestSuites/Validate/hub (0.01s)
        --- PASS: TestSuites/Validate/c1 (0.01s)
        --- PASS: TestSuites/Validate/c2 (0.01s)
    --- FAIL: TestSuites/Exhaustive (6.04s)
        --- FAIL: TestSuites/Exhaustive/subscr-deploy-rbd-busybox (0.01s)
            --- FAIL: TestSuites/Exhaustive/subscr-deploy-rbd-busybox/Undeploy (0.01s)
        --- FAIL: TestSuites/Exhaustive/appset-deploy-rbd-busybox (0.01s)
            --- FAIL: TestSuites/Exhaustive/appset-deploy-rbd-busybox/Undeploy (0.01s)
        --- FAIL: TestSuites/Exhaustive/appset-deploy-cephfs-busybox (0.01s)
            --- FAIL: TestSuites/Exhaustive/appset-deploy-cephfs-busybox/Undeploy (0.01s)
        --- FAIL: TestSuites/Exhaustive/subscr-deploy-cephfs-busybox (0.01s)
            --- FAIL: TestSuites/Exhaustive/subscr-deploy-cephfs-busybox/Undeploy (0.01s)
        --- PASS: TestSuites/Exhaustive/disapp-deploy-cephfs-busybox (2.52s)
            --- PASS: TestSuites/Exhaustive/disapp-deploy-cephfs-busybox/Undeploy (2.52s)
        --- PASS: TestSuites/Exhaustive/disapp-deploy-rbd-busybox (8.53s)
            --- PASS: TestSuites/Exhaustive/disapp-deploy-rbd-busybox/Undeploy (8.53s)
FAIL
```

Undeploy must be idempotent, if resources is not found we should log a debug message and continue.

Required for ramenctl.

--- a/e2e/deployers/appset.go
+++ b/e2e/deployers/appset.go
@@ -7,6 +7,7 @@ import (
 	"github.com/ramendr/ramen/e2e/config"
 	"github.com/ramendr/ramen/e2e/types"
 	"github.com/ramendr/ramen/e2e/util"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
 type ApplicationSet struct{}
@@ -58,11 +59,16 @@ func (a ApplicationSet) Undeploy(ctx types.Context) error {
 
 	clusterName, err := util.GetCurrentCluster(util.Ctx.Hub, managementNamespace, name)
 	if err != nil {
-		return err
-	}
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
 
-	log.Infof("Undeploying applicationset app \"%s/%s\" in cluster %q",
-		ctx.AppNamespace(), ctx.Workload().GetAppName(), clusterName)
+		log.Debugf("Could not retrieve the cluster name: %s", err)
+		log.Infof("Undeploying applicationset app \"%s/%s\"", ctx.AppNamespace(), ctx.Workload().GetAppName())
+	} else {
+		log.Infof("Undeploying applicationset app \"%s/%s\" in cluster %q",
+			ctx.AppNamespace(), ctx.Workload().GetAppName(), clusterName)
+	}
 
 	err = DeleteApplicationSet(ctx, a)
 	if err != nil {

--- a/e2e/deployers/subscr.go
+++ b/e2e/deployers/subscr.go
@@ -7,6 +7,7 @@ import (
 	"github.com/ramendr/ramen/e2e/config"
 	"github.com/ramendr/ramen/e2e/types"
 	"github.com/ramendr/ramen/e2e/util"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	subscriptionv1 "open-cluster-management.io/multicloud-operators-subscription/pkg/apis/apps/v1"
 )
 
@@ -80,11 +81,16 @@ func (s Subscription) Undeploy(ctx types.Context) error {
 
 	clusterName, err := util.GetCurrentCluster(util.Ctx.Hub, managementNamespace, name)
 	if err != nil {
-		return err
-	}
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
 
-	log.Infof("Undeploying subscription app \"%s/%s\" in cluster %q",
-		ctx.AppNamespace(), ctx.Workload().GetAppName(), clusterName)
+		log.Debugf("Could not retrieve the cluster name: %s", err)
+		log.Infof("Undeploying subscription app \"%s/%s\"", ctx.AppNamespace(), ctx.Workload().GetAppName())
+	} else {
+		log.Infof("Undeploying subscription app \"%s/%s\" in cluster %q",
+			ctx.AppNamespace(), ctx.Workload().GetAppName(), clusterName)
+	}
 
 	err = DeleteSubscription(ctx, s)
 	if err != nil {

--- a/e2e/dractions/actions.go
+++ b/e2e/dractions/actions.go
@@ -8,6 +8,7 @@ import (
 	"github.com/ramendr/ramen/e2e/config"
 	"github.com/ramendr/ramen/e2e/types"
 	"github.com/ramendr/ramen/e2e/util"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/client-go/util/retry"
 )
 
@@ -108,11 +109,16 @@ func DisableProtection(ctx types.Context) error {
 
 	clusterName, err := util.GetCurrentCluster(util.Ctx.Hub, managementNamespace, placementName)
 	if err != nil {
-		return err
-	}
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
 
-	log.Infof("Unprotecting workload \"%s/%s\" in cluster %q",
-		appNamespace, ctx.Workload().GetAppName(), clusterName)
+		log.Debugf("Could not retrieve the cluster name: %s", err)
+		log.Infof("Unprotecting workload \"%s/%s\"", appNamespace, ctx.Workload().GetAppName())
+	} else {
+		log.Infof("Unprotecting workload \"%s/%s\" in cluster %q",
+			appNamespace, ctx.Workload().GetAppName(), clusterName)
+	}
 
 	drpcName := name
 

--- a/e2e/dractions/disapp.go
+++ b/e2e/dractions/disapp.go
@@ -9,6 +9,7 @@ import (
 	"github.com/ramendr/ramen/e2e/deployers"
 	"github.com/ramendr/ramen/e2e/types"
 	"github.com/ramendr/ramen/e2e/util"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
 func EnableProtectionDiscoveredApps(ctx types.Context) error {
@@ -73,11 +74,16 @@ func DisableProtectionDiscoveredApps(ctx types.Context) error {
 
 	clusterName, err := util.GetCurrentCluster(util.Ctx.Hub, managementNamespace, placementName)
 	if err != nil {
-		return err
-	}
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
 
-	log.Infof("Unprotecting workload \"%s/%s\" in cluster %q",
-		appNamespace, ctx.Workload().GetAppName(), clusterName)
+		log.Debugf("Could not retrieve the cluster name: %s", err)
+		log.Infof("Unprotecting workload \"%s/%s\"", appNamespace, ctx.Workload().GetAppName())
+	} else {
+		log.Infof("Unprotecting workload \"%s/%s\" in cluster %q",
+			appNamespace, ctx.Workload().GetAppName(), clusterName)
+	}
 
 	if err := deleteDRPC(ctx, util.Ctx.Hub, managementNamespace, drpcName); err != nil {
 		return err

